### PR TITLE
Refactor mainly css-styling

### PR DESCRIPTION
**Pull request by aruppert**

### CODE REVIEW

File: client/src/api/questions.js [hunk 1]
```diff
@@ -1,19 +1,5 @@
...
-  const questionData = {
-    question,
-    correct_answer,
-    incorrect_answer1,
-    incorrect_answer2,
-    incorrect_answer3,
-    status
-  };
+export function addQuestion(questionData) {
+  console.log(questionData);
```
Comment (marwinburesch) [inline client/src/api/questions.js]: you can leave this line out :)

File: client/src/components/AddQuestionsCard.js [hunk 2]
```diff
@@ -2,34 +2,25 @@ import React from 'react';
 import styled from '@emotion/styled';
 import AddQuestion from '../icons/AddQuestion';
 import { Link } from 'react-router-dom';
+import { flexColumnCenter, linearGradientBoxShadow, TextCenterColorOneMargin20 } from '../styles/General';
 
 const AddQuestionsContainer = styled(Link)`
-  display: flex;
-  flex-flow: column;
+  ${flexColumnCenter};
+  ${linearGradientBoxShadow};
```
Comment (marwinburesch) [inline client/src/components/AddQuestionsCard.js]: great idea!

File: client/src/App.js [hunk 3]
```diff
@@ -32,45 +35,54 @@ function App() {
...
+                    }
+                    amountOfQuestions={amountOfQuestions}
+                    setNumberOfPlayers={number => setNumberOfPlayers(number)}
+                    numberOfPlayers={numberOfPlayers}
+                    setPrivateCode={value => setPrivateCode(value)}
+                    chooseNamePlayer1={chooseNamePlayer1}
+                    chooseNamePlayer2={chooseNamePlayer2}
+                  />
+                </Route>
+                <Route path="/privacy">
```
Comment (marwinburesch) [inline client/src/App.js]: I'd call it `private`

File: client/src/components/AddQuestionsCard.js [hunk 4]
```diff
@@ -2,34 +2,25 @@ import React from 'react';
...
-  text-align: center;
-  align-self: center;
-  color: ${props => props.theme.colors.text1};
+  ${TextCenterColorOneMargin20};
 `;
 
 export default function AddQuestionsCard() {
   return (
-    <AddQuestionsContainer to="/puborprivate">
+    <AddQuestionsContainer to="/privacy">
```
Comment (marwinburesch) [inline client/src/components/AddQuestionsCard.js]: see `App.js` line 58

File: client/src/components/AnswerCard.js [hunk 5]
```diff
@@ -3,13 +3,13 @@ import styled from '@emotion/styled';
 
 const AnswerContainer = styled.button`
   display: flex;
```
Comment (marwinburesch) [inline client/src/components/AnswerCard.js]: I think you can center the content of `<button>` by using `margin: 0 auto` on its children. You should consider checking all your elements for `display: flex` and see if you could get child elements centred via `margin`

File: client/src/components/AnswerCard.js [hunk 6]
```diff
@@ -3,13 +3,13 @@ import styled from '@emotion/styled';
...
-  outline: none;
-  border: 4px solid ${props => props.theme.colors.text1};
   justify-content: center;
+  outline: none;
   margin: 10px;
   width: 160px;
   height: 90px;
```
Comment (marwinburesch) [inline client/src/components/AnswerCard.js]: I'm not sure about fixed `width` & `height`, might be worth considering a separate overhaul for responsiveness if you find the time to do so ;)

File: client/src/components/Footer.js [hunk 7]
```diff
@@ -4,60 +4,44 @@ import Dice from '../icons/Dice';
...
 const FooterBar = styled.div`
-  position: absolute;
-  bottom: 0px;
+  ${linearGradientBoxShadow};
   display: flex;
   justify-content: space-around;
   align-items: center;
-  position: absolute;
   bottom: 0px;
+  position: absolute;
```
Comment (marwinburesch) [inline client/src/components/Footer.js]: The `FooterBar` doesn't have to have `position: absolute`, if the parent element of main layout elements (such as `main`, `header` and `footer`) is a flexbox you can give `footer` and `header` a fixed height and the `main` element a `flex-grow: 1`

see `main`, `header` and `footer` on this [exercise](https://github.com/marwinburesch/time.wise/blob/solution-html/style.scss ) for an example.

File: client/src/pages/ChooseGameModePage.js [hunk 8]
```diff
@@ -1,43 +1,32 @@
...
-    to right,
-    ${props => props.theme.colors.card2},
-    ${props => props.theme.colors.card1}
-  );
-  border-radius: 25px 0px 25px 0px;
-  box-shadow: 0px 3px 6px rgba(0, 0, 0, 0.16);
   text-align: center;
-  align-self: center;
   color: ${props => props.theme.colors.text1};
+  width: 340px;
```
Comment (marwinburesch) [inline client/src/pages/ChooseGameModePage.js]: again, you should consider using `%` as unit here for easy "responsiveness"

File: client/src/pages/FormPage.js [hunk 9]
```diff
@@ -1,24 +1,237 @@
...
+    incorrect_answer1: '',
+    incorrect_answer2: '',
+    incorrect_answer3: '',
+    privateCode: privateCode
+  });
+  const [questionStatus, setQuestionStatus] = React.useState('inactive');
+  const [questionsCounter, setQuestionsCounter] = React.useState(0);
+  const [finished, setFinished] = React.useState(false);
+  const [showError, setShowError] = React.useState(false);
+  console.log(privateCode);
```
Comment (marwinburesch) [inline client/src/pages/FormPage.js]: 🌲save the forest, less logging!

File: client/src/styles/General.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
+import { css } from '@emotion/core';
+
+export const linearGradientBoxShadow = props => css`
```
Comment (marwinburesch) [inline client/src/styles/General.js]: mmmmh `props` ;) 👌

File: client/src/App.js (hunk 3 above)
Comment (aruppert) [reply client/src/App.js]: but it can be public too ;)

File: client/src/components/AddQuestionsCard.js (hunk 2 above)
Comment (aruppert) [reply client/src/components/AddQuestionsCard.js]: thanks to Leon :)

File: client/src/components/AnswerCard.js (hunk 5 above)
Comment (aruppert) [reply client/src/components/AnswerCard.js]: Thanks, will do next round refactoring!

File: client/src/components/AnswerCard.js (hunk 6 above)
Comment (aruppert) [reply client/src/components/AnswerCard.js]: Yeah, I was thinking about that too

File: client/src/components/Footer.js (hunk 7 above)
Comment (aruppert) [reply client/src/components/Footer.js]: true, but is there any downside of it?

File: client/src/pages/ChooseGameModePage.js (hunk 8 above)
Comment (aruppert) [reply client/src/pages/ChooseGameModePage.js]: see above. but thanks again!

File: client/src/pages/FormPage.js (hunk 9 above)
Comment (aruppert) [reply client/src/pages/FormPage.js]: Hehe, will make a habit of searching for those climate killers next time before push ;)

File: client/src/styles/General.js (hunk 10 above)
Comment (aruppert) [reply client/src/styles/General.js]: :)

File: client/src/api/questions.js (hunk 1 above)
Comment (aruppert) [reply client/src/api/questions.js]: oops

File: client/src/components/Footer.js (hunk 7 above)
Comment (marwinburesch) [reply client/src/components/Footer.js]: Not really, unless it doesn't fit your app

**Pull request merged**